### PR TITLE
use fragment instead of query in jetbrains gateway links

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

It turned out that gateway links expect parameters to be encoded in fragment not query. It worked somehow, but was not correct and broke for some clients.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/7965

## How to test

- Start a new workspace: https://ak-fix-jb-gw-links.staging.gitpod-dev.com/#referrer:jetbrains-gateway/https://github.com/open-vsx/publish-extensions
- Make sure that Gateway can connect to it.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **akosyakov** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **akosyakov** after the PR has been reviewed.

Associated issue: *#7965*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jeanp413 — APPROVED** [review]
/lgtm
I could connect successfully with jb gateway and gitpod extension installed on Mac

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: d825ba4bbc0ad53c249b0a7223579bbd6b6c006c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7952#pullrequestreview-870526272" title="LGTM">jeanp413</a>*

Associated issue: *#7965*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
